### PR TITLE
feat(nx-plugin): ensureNxProject uses project names and resolves workspace dependencies

### LINKED ISSUE #4851: @nrwl/nx-plugin:e2e does not support plugins using paths from tsconfig.base.json
## Current Behavior
When importing within the same workspace plugins e2e fail.

## Expected Behavior
I should be able to run e2e tests in plugins workspace when importing within the same workspace.

## Steps to Reproduce
1. Create new nrwl workspace: `npx create-nx-plugin my-org --pluginName core`
2. Create second plugin: `nx g @nrwl/nx-plugin:plugin serverless`
3. In `core` create some utility method
4. Use it in `serverless`
5. Run `nx e2e serverless-e2e`

[Repository link](https://github.com/Bielik20/nx-plugins/tree/ff1b00e42d605cba38979c1b3d1125b4652cad24).

### Failure Logs

```
    Command failed: npm install --legacy-peer-deps

      at runPackageManagerInstall (../../packages/nx-plugin/src/utils/testing-utils/nx-project.ts:50:19)
      at newNxProject (../../packages/nx-plugin/src/utils/testing-utils/nx-project.ts:70:3)
      at Object.ensureNxProject (../../packages/nx-plugin/src/utils/testing-utils/nx-project.ts:82:3)
      at tests/serverless.spec.ts:37:7
      at ../../node_modules/tslib/tslib.js:117:75
      at Object.__awaiter (../../node_modules/tslib/tslib.js:113:16)
```

### Environment

```
  Node : 12.18.3
  OS   : darwin x64
  npm  : 6.14.6

  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 11.3.0
  @nrwl/cypress : Not Found
  @nrwl/devkit : 11.3.0
  @nrwl/eslint-plugin-nx : 11.3.0
  @nrwl/express : Not Found
  @nrwl/jest : 11.3.0
  @nrwl/linter : 11.3.0
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : 11.3.0
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 11.3.0
  @nrwl/web : Not Found
  @nrwl/workspace : 11.3.0
  typescript : 4.0.7
```

### PR DESCRIPTION
**Pull request by samuelfernandez**
## Current Behavior
`ensureNxProject` requires a package name and dist folder. This doesn't work well when a project has dependencies within the workspace.

## Expected Behavior

`ensureNxProject` will now support a list of projects, and will also resolve transient dependencies within the workspace.

## Related Issue(s)

Fixes #4851

### CODE REVIEW

File: packages/devkit/src/utils/module-federation/dependencies.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ function collectDependencies(
   return dependencies;
 }
 
-function getLibraryImportPath(
+export function getLibraryImportPath(
```
Comment (AgentEnder) [inline packages/devkit/src/utils/module-federation/dependencies.ts:56]: Let's not export this - I don't think devkit's public api should change for this.

**Comment by samuelfernandez** [comment]
Still missing the migration generator

File: packages/nx-plugin/migrations.json [hunk 2]
```diff
@@ -29,6 +29,12 @@
       "cli": "nx",
       "description": "Update nx plugin jest test files to support jest 29 changes (https://jestjs.io/docs/upgrading-to-jest29)",
       "factory": "./src/migrations/update-15-9-0/jest-29-tests"
+    },
+    "update-nx-plugin-e2e-tests": {
+      "version": "15.9.3",
```
Comment (samuelfernandez) [inline packages/nx-plugin/migrations.json:34]: @AgentEnder review this version before mergin

File: packages/devkit/src/utils/module-federation/dependencies.ts (hunk 1 above)
Comment (samuelfernandez) [reply packages/devkit/src/utils/module-federation/dependencies.ts:56]: Will remove it from `public-api.ts`, yet need to keep the export for internal usage

**Comment by samuelfernandez** [comment]
@AgentEnder this is ready to review, including migration! 😄

File: packages/nx-plugin/migrations.json (hunk 2 above)
Comment (AgentEnder) [inline packages/nx-plugin/migrations.json:34]: Lets target the v16 release for this - 16.0.0-beta.1 should be fine.

File: packages/nx-plugin/src/migrations/update-15-9-3/__snapshots__/update-nx-plugin-e2e-tests.spec.ts.snap
Comment (AgentEnder) [inline packages/nx-plugin/src/migrations/update-15-9-3/__snapshots__/update-nx-plugin-e2e-tests.spec.ts.snap:1]: We just merged an update that bumps us to jest snapshot v2. Can you rebase and update the snapshots?

File: packages/nx/src/project-graph/utils/get-output-path.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+export function getOutputPath(
+  projectGraph: ProjectGraph,
+  name: string
+): string {
+  const node = projectGraph.nodes[name];
+  if (node == null) {
+    throw new Error(`Could not find project with name "${name}"`);
+  }
+
+  const buildTarget = node.data.targets?.build;
```
Comment (AgentEnder) [inline packages/nx/src/project-graph/utils/get-output-path.ts:12]: can we do something like:
```typescript
Object.values(node.data.targets ?? {})
	.find(x => [
			'@nrwl/js:tsc', 
			'@nrwl/js:swc'
		].includes(x.executor)
	) ?? node.data.targets??.build
```

**Review by AgentEnder — CHANGES_REQUESTED** [review]
Looking good! Few changes:

**Comment by AgentEnder** [comment]
@samuelfernandez I really appreciate the effort you put in on the PR for this, but we are working on a slightly different more complex approach to solving this problem. 

Since we are going in a different direction, I'm going to close this out. If I had known sooner, I would have let you know. Apologies....

Stay tuned. 📺

**Pull request closed without merging**